### PR TITLE
define charset for  .*, \w, \W, \d, \D, \s, \S

### PR DESCRIPTION
**Pull request by tomgregar**
updated tests (defined charset for .*, \w, \W, \d, \D, \s, \S)
updated xeger (definition of that charset)
updated regexp - use that set of chars instead of "printable ascii" used before + create function for print usedcharset for any regexp

### CODE REVIEW

File: Src/Fare/RegExp.cs [hunk 1]
```diff
@@ -71,8 +73,8 @@ private RegExp()
         ///   Initializes a new instance of the <see cref = "RegExp" /> class from a string.
         /// </summary>
         /// <param name = "s">A string with the regular expression.</param>
-        public RegExp(string s)
-            : this(s, RegExpSyntaxOptions.All)
+        public RegExp(string s, string anyCharCharset = null)
```
Comment (Licho1) [inline Src/Fare/RegExp.cs]: docs, what is meaning of "anycharcharset"

File: Src/Fare/RegExp.cs [hunk 2]
```diff
@@ -634,6 +649,95 @@ private StringBuilder ToStringBuilder(StringBuilder sb)
             return sb;
         }
 
+        public string UsedChars()
+        {
+            var sb1 = new StringBuilder();
+            return this.ToAlphabetStringBuilder(sb1).ToString();
+        }
+        private StringBuilder ToAlphabetStringBuilder(StringBuilder sb)
```
Comment (Licho1) [inline Src/Fare/RegExp.cs]: what is this thing for /doing?

Comment (tomgregar) [reply Src/Fare/RegExp.cs]: to create charset from regex. for usage with patternelement (or regexelement) to check what chars can be there in code for generate train data for ocr training.

File: Src/Fare/RegExp.cs (hunk 1 above)
Comment (tomgregar) [reply Src/Fare/RegExp.cs]: ok

**Review by Licho1 — APPROVED** [review]

**Pull request merged**